src/infrastructure/channels/mod.rs

```diff
@@ -0,0 +1,24 @@
+///! This module provides channels for processing packets of different reliabilities.
+
+mod unreliable_channel;
+mod sequenced_channel;
+mod reliable_channel;
+
+use infrastructure::DeliveryMethod;
+use error::NetworkResult;
+use packet::PacketData;
+use std::io::Cursor;
+
+pub use self::unreliable_channel::UnreliableChannel;
+pub use self::sequenced_channel::SequencedChannel;
+pub use self::reliable_channel::ReliableChannel;
+
+/// This provides an abstraction for processing packets to their given reliability.
+pub trait Channel {
+    /// Process an packet before send and return an packet data instance with the given raw packet data.
+    fn pre_process(&mut self, payload: &[u8], delivery_method: DeliveryMethod) -> NetworkResult<PacketData>;
```

Comment:
Do we need to pass a delivery_method? I'm not sure I want to think about this a bit more.

Reply:
For us in order to process the packet, we can pass a reference to the `Packet` used some levels above which also has the `delivery_method` but we need to know the `delivery_type` in order to pre-process packets.